### DISCUSSION TITLE
Custom reusable component

### DISCUSSION THREAD

**Discussion by samar-elsayed** [apache/camel-karavan#1279] [Q&A] [answered]
### Describe the solution you'd like

there is a use case where we might repeat certain route steps in all / many karavan projects with the same content / with some variables

is there a low code way to create a central component that groups these steps, **acts as a template**, and **can accept variables as input**? 

so that when we need to make some changes in these steps, we will be able to make them inside one place (the component) and see them reflected in all projects using this component  

### Variant

Vs Code Extension

### Container Management (if applicable)

Kubernetes

**Comment by davsclaus** [comment, marked as answer]
Take a look at kamelets
